pg/src/main/java/org/bouncycastle/openpgp/PGPPublicKeyRing.java

```diff
@@ -338,12 +338,12 @@ public static PGPPublicKeyRing removePublicKey(
         PGPPublicKeyRing  pubRing,
         PGPPublicKey      pubKey)
     {
-        List       keys = new ArrayList(pubRing.keys);
+        List<PGPPublicKey> keys = new ArrayList<>(pubRing.keys);
         boolean    found = false;
         
-        for (int i = 0; i < keys.size();i++)
```

Comment:
Again, I reversed direction of the for-loop to prevent ConcurrentModificationExceptions.

Reply:
The operation is on an internal list in an immutable class (well at least it's supposed to be). I think this might be a step further than necessary.

Reply:
Shall I update my PR removing those changes, or do you just pick whatever changes you like?

Reply:
I'm going okay just picking things up I think. I'll close this one when I think I'm done.